catalog/core/catalog-core-solr/src/main/java/ddf/catalog/source/solr/SolrMetacardClientImpl.java

```diff
@@ -267,21 +266,39 @@ private FacetAttributeResult convertFacetField(FacetField facetField) {
       QueryResponse solrResponse = client.query(query, METHOD.POST);
       SolrDocumentList docs = solrResponse.getResults();
 
-      List<Metacard> results = new ArrayList<>();
-      for (SolrDocument doc : docs) {
-        try {
-          results.add(createMetacard(doc));
-        } catch (MetacardCreationException e) {
-          throw new UnsupportedQueryException("Could not create metacard(s).", e);
-        }
-      }
+      List<Metacard> results = createMetacards(docs);
 
       return results;
     } catch (SolrServerException | SolrException | IOException e) {
       throw new UnsupportedQueryException("Could not complete solr query.", e);
     }
   }
 
+  @Override
+  public List<Metacard> getIds(Set<String> ids) throws UnsupportedQueryException {
+    try {
+      SolrDocumentList docs = client.getById(ids);
+
+      List<Metacard> results = createMetacards(docs);
+
+      return results;
+    } catch (SolrServerException | SolrException | IOException e) {
+      throw new UnsupportedQueryException("Could not complete solr query.", e);
+    }
+  }
+
+  private List<Metacard> createMetacards(SolrDocumentList docs) throws UnsupportedQueryException {
+    List<Metacard> results = new ArrayList<>();
```

Comment:
❓ Have you considered initializing the array list with the same number of elements as SolrDocumentList such that there would be no need to allocate more memory then required or go through multiple resizing if more is required than the default?